saw-central/src/SAWCentral/Crucible/Common/Pred.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module SAWCentral.Crucible.Common.Pred (
+    checkBooleanType
+    ) where
```

Comment:
Rather than making a separate module for this, I propose putting this in the existing `SAWCentral.Crucible.Common.ResolveSetupValue` module.